test/docx/body-reader.tests.js

```diff
@@ -94,6 +94,23 @@ test("paragraph has justification read from paragraph properties if present", fu
     assert.deepEqual(paragraph.alignment, "center");
 });
 
+var indentProperties = [
+    {name: "left", value: {"w:left": "720"}},
+    {name: "right", value: {"w:right": "180"}},
+    {name: "firstLine", value: {"w:firstLine": "320"}},
+    {name: "hanging", value: {"w:hanging": "500"}}
+];
+
+indentProperties.forEach(function(property) {
+    test("paragraph has indent " + property.name + " read from paragraph properties if present", function() {
+        var indentXml = new XmlElement("w:ind", property.value, []);
+        var propertiesXml = new XmlElement("w:pPr", {}, [indentXml]);
+        var paragraphXml = new XmlElement("w:p", {}, [propertiesXml]);
+        var paragraph = readXmlElementValue(paragraphXml);
+        assert.deepEqual(paragraph.indent[property.name], property.value[Object.keys(property.value)[0]]);
```

Comment:
This feels like over-complicated logic for a test.

Perhaps instead something like:

```javascript
test("paragraph has indent left read from paragraph properties if present", function() {
    var paragraphXml = paragraphWithIndent({"w:left": "720"});
    var paragraph = readXmlElementValue(paragraphXml);
    assert.equal(paragraph.indent.left, "720");
});

test("paragraph has indent right read from paragraph properties if present", function() {
    var paragraphXml = paragraphWithIndent({"w:right": "720"});
    var paragraph = readXmlElementValue(paragraphXml);
    assert.equal(paragraph.indent.right, "720");
});

// ...etc...

function paragraphWithIndent(indentAttributes) {
    var indentXml = new XmlElement("w:ind", indentAttributes, []);
    var propertiesXml = new XmlElement("w:pPr", {}, [indentXml]);
    return new XmlElement("w:p", {}, [propertiesXml]);
}
```

?

Reply:
It'd also be good to test for the case where the attribute is missing, and when the entire indent element is missing.

Reply:
Implemented your style of tests.
I can't test for when the indent element is missing. 
It will always be present, because of how the Paragraph object is built. But it's properties `left`, `right`, `firstLine`, and `hanging` will be null.

What do you suggest?

Reply:
Hello @mwilliamson, I have tried implementing the tests, but they keep failing. Any idea what I am missing?

Reply:
`p.indent` is never set, so you're trying to assign properties to `undefined`. I'd suggest avoiding mutation altogether and building the object in one go.

Reply:
I think I have added everything as requested

Reply:
I am not sure how to do it without mutation.
There are several validation tests needing to be done. 
If indent doesn't exist in source then it won't exists in the object as well.
It will only show up in paragraph object if it has some values.

Or are all properties supposed to exist in the object even if the property is not present in source?